File: free/CreateTargetArray.js
/**
 * Given two number arrays called nums and index, create the target array following a couple rules.
 * From left to right, insert at index index[i] the value nums[i] into target array.
 * Seem like you select nums[i] and place into index[i] for the target arr, which means it might be useful to fill the target arr with 0's already.
 * 
 * @param {number[]} nums
 * @param {number[]} index
 * @return {number[]}
 */
const createTargetArray = (nums, index) => {
	let target = [];

	for (let i = 0; i < nums.length; i++) {
		target.splice(index[i], 0, nums[i]);
	}

	return target;
};

File: golang/create_target_array.go
// create the target array given nums and index arrays.
func createTargetArray(nums []int, index []int) []int {
	target := []int{}

	for i := 0; i < len(nums); i++ {
		if index[i] == len(target) {
			target = append(target, nums[i])
		} else {
			target = append(target[:index[i]+1], target[index[i]:]...)
			target[index[i]] = nums[i]
		}
	}

	return target
}